jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/PrivateMessageController.java

```diff
@@ -242,18 +242,23 @@ public ModelAndView editDraftPage(@PathVariable(PM_ID) Long id) throws NotFoundE
      * @return redirect to "drafts" folder if saved successfully or show form with error message
      */
     @RequestMapping(value = "/pm/save", method = {RequestMethod.POST, RequestMethod.GET})
-    public String saveDraft(@Valid @ModelAttribute PrivateMessageDraftDto pmDto, BindingResult result) {
+    public ModelAndView saveDraft(@Valid @ModelAttribute("privateMessageDto") PrivateMessageDraftDto pmDto, BindingResult result) {
         String targetView = "redirect:/drafts";
-        if (result.hasFieldErrors() && result.hasGlobalErrors()) {
-            if (pmDto.getId() != 0) { //means that we try to edit existing draft
+        long pmDtoId = pmDto.getId();
+        if (result.hasGlobalErrors()) {
+            // It means that we have some ObjectErrors. It may occur due annotations placed over class.
+            if (pmDtoId != 0) { //means that we try to edit existing draft
                 try {
-                    pmService.delete(Arrays.asList(pmDto.getId()));
+                    pmService.delete(Arrays.asList(pmDtoId));
                 } catch (NotFoundException e) {
                     // Catch block is empty because we don't need any additional logic in case if user removed
```

Comment:
Are you sure we should keep this logic? Could you ask QA team about their opinion?